tools/rvn/CommandLineApp.cs

```diff
@@ -51,6 +61,130 @@ public static int Run(string[] args)
+                    var certPassTuple = certPass.Value() ?? Environment.GetEnvironmentVariable("RVN_CERT_PASS");
+
+                    return await CreateSetupPackage(new CreateSetupPackageParameters
+                    {
+                        SetupInfoPath = setupParamVal,
+                        PackageOutputPath = packageOutPathVal,
+                        Command = cmd,
+                        Mode = modeVal,
+                        CertificatePath = certPathVal,
+                        CertPassword = certPassTuple,
+                        Progress = new SetupProgressAndResult(tuple =>
+                        {
+                            if (tuple.Message != null)
+                            {
+                                Console.WriteLine(tuple.Message);
+                            }
+
+                            if (tuple.Exception != null)
+                            {
+                                Console.Error.WriteLine(tuple.Exception.Message);
+                            }
+                        }),
+                        CancellationToken = token
+                    });
+                });
+            });
+        }
+
+        private static async Task<int> CreateSetupPackage(CreateSetupPackageParameters parameters)
+        {
+            byte[] zipFile;
+            SetupInfo setupInfo;
+
+            if (File.Exists(parameters.SetupInfoPath) == false)
+            {
+                return ExitWithError($"{nameof(parameters.SetupInfoPath)} does not exits", parameters.Command);
+            }
+
+            if (Path.GetExtension(parameters.PackageOutputPath)?.Equals(".zip", StringComparison.OrdinalIgnoreCase) == false)
+            {
+                return ExitWithError($"'{parameters.PackageOutputPath}' file name must end with ZIP extension", parameters.Command);
+            }
+
+            try
+            {
+                using (StreamReader file = File.OpenText(parameters.SetupInfoPath))
+                {
+                    JsonSerializer serializer = new();
+                    setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
+                }
+            }
+            catch (Exception e)
+            {
+                return ExitWithError($"Failed to load RavenDB Setup parameters from file: {parameters.SetupInfoPath}{Environment.NewLine}Error: {e}", parameters.Command);
+            }
+
+            if (string.IsNullOrEmpty(setupInfo?.Domain))
+                return ExitWithError($"{nameof(setupInfo.Domain)} cannot be empty", parameters.Command);
+
+            parameters.PackageOutputPath ??= setupInfo.Domain;
+
+            if (string.IsNullOrEmpty(parameters.CertPassword) == false)
+            {
+                setupInfo.Password = parameters.CertPassword;
+            }
+
+            switch (parameters.Mode)
+            {
+                case OwnCertificate when string.IsNullOrEmpty(parameters.CertificatePath):
```

Comment:
One last thing. 
Can we move validation pieces outside of this mode switch? We should have 3 clear cases:
OwnCertificate (no ifs), 
LetsEncrypt (no ifs), 
default (ExitWithError("Invalid mode provided").